### DISCUSSION TITLE
GitOps dedicated "Conventional Commits" types

### DISCUSSION THREAD

**Discussion by mdufourneaud** [open-gitops/project#187] [Ideas]
I am wondering if there is a mapping between the "standard" conventional commit types and the GitOps operations, or if we need to add new types dedicated to GitOps.
In the 2nd case, here are the GitOps types I propose:
* `create:` a commit of the type `create` create new resources
* `update:` a commit of the type `update` update existing resources
  * `upgrade:` a commit of the type `upgrade`, a special update, upgrade existing resources
* `delete:` a commit of the type `delete` delete existing resources

**Comment by franzudev** [comment]
A well written commit message respecting the convetional commits standard will include these verbs in it's message, imho

**Comment by roberthstrand** [comment]
I like the idea as a concept, but at the same time I have been using conventional commits for a while with GitOps without thinking too much about it. However, for the sake of discussion, this makes a lot of sense for both infrastructure- and configuration-as-code.

I see that you have added an issue to the CC-repository, but I would personally change the scope of the request to IaC rather than just GitOps.

**Comment by cwstrommer** [comment]
What would be the purpose of defining these commit types?  And what is the scope - direct changes, or direct+dependencies?
